### PR TITLE
Feature/#335 make mypy happy

### PR DESCRIPTION
**Pull request by toan-quach**
#335

### CODE REVIEW

File: src/taipy/core/config/checkers/_data_node_config_checker.py [hunk 1]
```diff
@@ -22,7 +24,7 @@ def __init__(self, config: _Config, collector: IssueCollector):
         super().__init__(config, collector)
 
     def _check(self) -> IssueCollector:
-        data_node_configs = self._config._sections[DataNodeConfig.name]
+        data_node_configs: Dict[str, DataNodeConfig] = self._config._sections[DataNodeConfig.name]
         for data_node_config_id, data_node_config in data_node_configs.items():
```
Comment (toan-quach) [inline src/taipy/core/config/checkers/_data_node_config_checker.py:28]: @jrobinAV 
This file has several failed mypy tests in line 61, 70, 78, 93, 94, 110 with errors related to Optional[Any] (I believe it's related to the variable data_node_config used in each function), can't explain the problem though. Can you have a look?

**Review by jrobinAV — APPROVED** [review]

File: src/taipy/core/config/checkers/_data_node_config_checker.py (hunk 1 above)
Comment (jrobinAV) [reply src/taipy/core/config/checkers/_data_node_config_checker.py:28]: Apparently, mypy detects that data_node_config.properties can be None. It is built from **properties in the Section class in the taipy-config repo. when no kwargs are passed, then properties is not an empty dict but None. And so mypy tells us that some functions like "in" or "indexation" are incompatible with None type.

I just pushed a fix in the taipy-config repo. It should be solved.

**Review by jrobinAV — DISMISSED** [review]

File: src/taipy/core/config/checkers/_data_node_config_checker.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/core/config/checkers/_data_node_config_checker.py:28]: @jrobinAV 
I have tried running again but the problem is still there. I also tried adding type Dict to the @property properties in Section with no help at all!

Comment (jrobinAV) [reply src/taipy/core/config/checkers/_data_node_config_checker.py:28]: I don't have it anymore. Are you sure you pulled the latest develop version of taipy-config?

Comment (jrobinAV) [reply src/taipy/core/config/checkers/_data_node_config_checker.py:28]: OK, I finally fixed the issue in the current PR as well. Can you try again?

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**